### PR TITLE
Develop branch adapts to openharmony

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog
Adaptation modules.
1、File
2、Audio
3、Network
4、Project
5、Font
etc.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by qiuguohua** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/manual/jsb_global.cpp [hunk 1]
```diff
@@ -804,7 +808,7 @@ static bool JSB_setPreferredFramesPerSecond(se::State &s) { // NOLINT
 }
 SE_BIND_FUNC(JSB_setPreferredFramesPerSecond)
 
-#if CC_USE_EDITBOX
+#if CC_USE_EDITBOX
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_global.cpp]: extra space

File: scripts/native-pack-tool/package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "iconv-lite": "^0.4.24",
     "xxtea-node": "^1.1.4",
     "yargs": "^17.5.1",
-    "zlib": "^1.0.5"
+    "zlib": "^1.0.5",
+    "json5":"^2.2.1"
```
Comment (minggo) [inline scripts/native-pack-tool/package.json:21]: @PPpro please help to review it too.

File: templates/openharmony/AppScope/resources/base/element/string.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "string": [
+    {
+      "name": "app_name",
+      "value": "OpenHarmonyDemo"
```
Comment (minggo) [inline templates/openharmony/AppScope/resources/base/element/string.json]: OpenHarmonyDemo -> OpenHarmonyApp

File: templates/openharmony/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+    "license": "ISC",
```
Comment (minggo) [inline templates/openharmony/package.json:2]: What's the license?

File: native/cocos/bindings/manual/jsb_global.cpp (hunk 1 above)
Comment (qiuguohua) [reply native/cocos/bindings/manual/jsb_global.cpp]: OK

File: templates/openharmony/package.json (hunk 4 above)
Comment (qiuguohua) [reply templates/openharmony/package.json:2]: This is the default license generated by DecEco when creating a new project

File: templates/openharmony/AppScope/resources/base/element/string.json (hunk 3 above)
Comment (qiuguohua) [reply templates/openharmony/AppScope/resources/base/element/string.json]: OK

File: native/cocos/bindings/manual/jsb_conversions.h [hunk 5]
```diff
@@ -1230,8 +1230,6 @@ inline bool nativevalue_to_se(const ccstd::vector<T> &from, se::Value &to, se::O
         if constexpr (!std::is_pointer<T>::value && is_jsb_object_v<T>) {
             auto *pFrom = ccnew T(from[i]);
             nativevalue_to_se(pFrom, tmp, ctx);
-            tmp.toObject()->clearPrivateData();
-            tmp.toObject()->setPrivateData(pFrom);
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_conversions.h:1234]: Do these two lines cause any issues?

File: native/cocos/network/HttpRequest.h [hunk 6]
```diff
@@ -241,8 +241,8 @@ class CC_DLL HttpRequest : public RefCounted {
         _timeoutInSeconds = timeoutInSeconds;
     }
 
-    inline float getTimeout() const {
-        return _timeoutInSeconds;
+    inline long getTimeout() const {
+        return static_cast<long>(_timeoutInSeconds);
```
Comment (dumganhar) [inline native/cocos/network/HttpRequest.h]: Why change the type of return value, which will break the compatibility?
_timeoutInSeconds could be a float value, for instance, 1.5f means 1.5 seconds.

File: cocos/serialization/deserialize-dynamic.ts [hunk 7]
```diff
@@ -253,7 +253,7 @@ function compileDeserializeNative (_self: _Deserializer, klass: CCClassConstruct
             if (prop === undefined) {
                 continue;
             }
-            if (!fastMode && typeof prop !== 'object') {
+            if (typeof prop !== 'object') {
```
Comment (PPpro) [inline cocos/serialization/deserialize-dynamic.ts]: why remove this fastMode

File: native/cocos/platform/interfaces/modules/Device.cpp [hunk 8]
```diff
@@ -30,9 +30,32 @@
...
 
+uint32_t Device::getInnerWidth() {
+    ISystemWindow* systemWindow = CC_GET_MAIN_SYSTEM_WINDOW();
+    CC_ASSERT_NOT_NULL(systemWindow);
+    auto viewSize = systemWindow->getViewSize();
+    return static_cast<uint32_t>(viewSize.width);
+}
+
+uint32_t Device::getInnerHeight() {
+    ISystemWindow* systemWindow = CC_GET_MAIN_SYSTEM_WINDOW();
```
Comment (dumganhar) [inline native/cocos/platform/interfaces/modules/Device.cpp:46]: ISystemWindow was bound, could we get ISystemWindow in JS directly?

File: scripts/native-pack-tool/package.json (hunk 2 above)
Comment (PPpro) [reply scripts/native-pack-tool/package.json:21]: it's OK to add json5 dependency

File: cocos/serialization/deserialize-dynamic.ts (hunk 7 above)
Comment (qiuguohua) [reply cocos/serialization/deserialize-dynamic.ts]: There is a problem with not removing this serializatio

File: native/cocos/platform/interfaces/modules/Device.cpp (hunk 8 above)
Comment (qiuguohua) [reply native/cocos/platform/interfaces/modules/Device.cpp:46]: Only the interface is exported, not the instance? This instance cannot be created directly.

File: native/cocos/bindings/manual/jsb_conversions.h (hunk 5 above)
Comment (qiuguohua) [reply native/cocos/bindings/manual/jsb_conversions.h:1234]: Not sure, there should be a problem before

File: native/cocos/network/HttpRequest.h (hunk 6 above)
Comment (qiuguohua) [reply native/cocos/network/HttpRequest.h]: I'll try

File: native/cocos/platform/interfaces/modules/Device.cpp (hunk 8 above)
Comment (dumganhar) [reply native/cocos/platform/interfaces/modules/Device.cpp:46]: How about using `root.mainWindow.width/height`?

File: cocos/serialization/deserialize-dynamic.ts (hunk 7 above)
Comment (dumganhar) [reply cocos/serialization/deserialize-dynamic.ts]: What problem?

File: native/cocos/platform/interfaces/modules/Device.cpp (hunk 8 above)
Comment (qiuguohua) [reply native/cocos/platform/interfaces/modules/Device.cpp:46]: If we can use this interface, can't window.innerWidth be deleted?

Comment (dumganhar) [reply native/cocos/platform/interfaces/modules/Device.cpp:46]: Should not delete `window.innerWidth` which is for keeping compatibility with `web`.
So where do you use `Device::getInnerHeight` after you exported `Device::getInnerHeight` to JS?

**Review by minggo — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Review by PPpro — APPROVED** [review]

**Pull request merged**